### PR TITLE
fix: Adding remove_unused_param for method and fixing same for assoc func

### PR DESCRIPTION
**Pull request by feniljain**
Solves #9571

### CODE REVIEW

File: crates/ide_assists/src/handlers/remove_unused_param.rs [hunk 1]
```diff
@@ -315,30 +350,23 @@ fn bar() {
...
+    fn test_remove_method_param() {
         check_assist(
             remove_unused_param,
             r#"
 struct S;
 impl S { fn f(&self, $0_unused: i32) {} }
 fn main() {
     S.f(92);
-    S.f();
-    S.f(92, 92);
```
Comment (matklad) [inline crates/ide_assists/src/handlers/remove_unused_param.rs:330]: Let's leave this in -- it's important to check that the code doesn't crash even if there are some type errors.

Comment (feniljain) [reply crates/ide_assists/src/handlers/remove_unused_param.rs:330]: So when I add these and run the tests, it works for the first case, as there's no arguments and it has nothing to do, but for the second case, i.e. S.f(92, 92), it removes the first argument, as it should normally do with correct cases.

**Comment by matklad** [comment]
that's correct behavior

**Pull request merged**